toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -276,16 +321,38 @@ func (im *IsoMaker) prepareWorkDirectory() {
 // copyStaticIsoRootFiles copies architecture-independent files from the
 // Mariner repo directories.
 func (im *IsoMaker) copyStaticIsoRootFiles() {
-	staticIsoRootFilesPath := filepath.Join(im.resourcesDirPath, "assets/isomaker/iso_root_static_files/*")
 
-	logger.Log.Debugf("Copying static ISO root files from '%s'.", staticIsoRootFilesPath)
+	if im.resourcesDirPath == "" && im.grubCfgPath == "" {
+		logger.Log.Panicf("Missing required parameters. Must specify either the resources directory or provide a grub.cfg.")
+	}
+
+	if im.resourcesDirPath != "" {
+		staticIsoRootFilesPath := filepath.Join(im.resourcesDirPath, "assets/isomaker/iso_root_static_files/*")
+
+		logger.Log.Infof("Copying static ISO root files from '%s' to '%s'.", staticIsoRootFilesPath, im.buildDirPath)
 
-	recursiveCopyDereferencingLinks(staticIsoRootFilesPath, im.buildDirPath)
+		recursiveCopyDereferencingLinks(staticIsoRootFilesPath, im.buildDirPath)
+	}
+
+	// im.grubCfgPath allows the user to overwrite the default grub.cfg that is
+	// copied from the resource folder.
+	if im.grubCfgPath != "" {
+		targetGrubCfg := filepath.Join(im.buildDirPath, "boot/grub2/grub.cfg")
+		targetGrubCfgDir := filepath.Dir(targetGrubCfg)
+		logger.PanicOnError(os.MkdirAll(targetGrubCfgDir, os.ModePerm), "Failed while creating directory '%s'.", targetGrubCfgDir)
+
+		logger.Log.Infof("Copying '%s' to '%s'.", im.grubCfgPath, targetGrubCfg)
+		shell.MustExecuteLive("cp", im.grubCfgPath, targetGrubCfg)
```

Comment:
Could we extract that to a separate function? Similar to `recursiveCopyDereferencingLinks`.

Reply:
My thinking is that this function copies a set of files, and then follows with some overrides - so, I kept them together. Both blocks of code deal with copying static architecture neutral files needed for the iso image. What do you think Pawel?

Reply:
Resolving. Plz, re-activate if needed.